core/math/color.cpp

```diff
@@ -109,35 +109,29 @@ uint64_t Color::to_rgba64() const {
 	return c;
 }
 
-String _to_hex(float p_val) {
+void _append_hex(float p_val, String &string) {
 	int v = Math::round(p_val * 255.0f);
 	v = CLAMP(v, 0, 255);
-	String ret;
 
 	for (int i = 0; i < 2; i++) {
-		char32_t c[2] = { 0, 0 };
 		int lv = v & 0xF;
 		if (lv < 10) {
```

Comment:
if we really wanted, we could define an array char hexa[] = { '0', ...., 'z'}; and just index into it, removing conditionals : ie : hexa[lv]. But that's minor stuff.

Reply:
(which would be good for cache misses because this is an unpredictable branch)

Reply:
Yeahh, more could be done but i wouldn't bother complicating the code over this. It's probably not even a bottleneck.
I mostly did this one separately because I wanted to get the implicit stolen use out of the way for an upcoming PR.

Reply:
Like I said, totally minor, just wanted to mention it for posterity, if someone ever want to try it one day.